src/pal/pal_open_enclave.h

```diff
@@ -9,6 +9,20 @@ extern "C" [[noreturn]] void oe_abort();
 
 namespace snmalloc
 {
+  // Use a template to hold the heap base and end variables.
+  // Templates prevent multiple-definition error for these variables,
+  // without having to use a .cpp file otherwise.
+  template<typename T>
+  struct HeapBounds
+  {
+    static std::atomic<void*> heap_base;

```

Comment:
We can use C++ inline statics here to remove the need for the template.
```C++
inline static std::atomic<void*> heap_base;
```

Reply:
I had to use `inline` variables instead of `inline static` so that I could initialize these in src/test/func/two_alloc_types/main.cc

Reply:
I think `inline static` would be required if you kept the struct. But the `inline` global is perfect here.  Thanks for addressing.